beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java

```diff
@@ -62,47 +68,54 @@ public SafeFuture<SendSignedBlockResult> sendSignedBlock(
         .thenPeek(performanceTracker::saveProducedBlock)
         .thenCompose(
-            signedBlock -> {
-              // creating blob sidecars after unblinding the block to ensure in the blinded flow we
-              // will have the cached builder payload
-              final List<BlobSidecar> blobSidecars =
-                  blockFactory.createBlobSidecars(blockContainer, blockPublishingPerformance);
-              return gossipAndImportUnblindedSignedBlockAndBlobSidecars(
-                  signedBlock, blobSidecars, broadcastValidationLevel, blockPublishingPerformance);
-            })
+            // creating blob sidecars after unblinding the block to ensure in the blinded flow we
+            // will have the cached builder payload
+            signedBlock ->
+                gossipAndImportUnblindedSignedBlockAndBlobSidecars(
+                    signedBlock,
+                    Suppliers.memoize(
+                        () ->
+                            blockFactory.createBlobSidecars(
+                                blockContainer, blockPublishingPerformance)),
+                    broadcastValidationLevel,
+                    blockPublishingPerformance))
         .thenCompose(result -> calculateResult(blockContainer, result, blockPublishingPerformance));
   }
 
   private SafeFuture<BlockImportAndBroadcastValidationResults>
       gossipAndImportUnblindedSignedBlockAndBlobSidecars(
           final SignedBeaconBlock block,
-          final List<BlobSidecar> blobSidecars,
+          final Supplier<List<BlobSidecar>> blobSidecars,
           final BroadcastValidationLevel broadcastValidationLevel,
           final BlockPublishingPerformance blockPublishingPerformance) {
 
     if (broadcastValidationLevel == BroadcastValidationLevel.NOT_REQUIRED) {
       // when broadcast validation is disabled, we can publish the block (and blob sidecars)
       // immediately and then import
-      publishBlockAndBlobSidecars(block, blobSidecars, blockPublishingPerformance);
-      return importBlockAndBlobSidecars(
-          block, blobSidecars, broadcastValidationLevel, blockPublishingPerformance);
+      publishBlock(block, blockPublishingPerformance)
+          .always(() -> publishBlobSidecars(blobSidecars.get(), blockPublishingPerformance));
+
+      importBlobSidecars(blobSidecars.get(), blockPublishingPerformance);
+      return importBlock(block, broadcastValidationLevel, blockPublishingPerformance);
     }
 
     // when broadcast validation is enabled, we need to wait for the validation to complete before
     // publishing the block (and blob sidecars)
 
     final SafeFuture<BlockImportAndBroadcastValidationResults>
         blockImportAndBroadcastValidationResults =
-            importBlockAndBlobSidecars(
-                block, blobSidecars, broadcastValidationLevel, blockPublishingPerformance);
+            importBlock(block, broadcastValidationLevel, blockPublishingPerformance);
```

Comment:
I feel it requires more refactoring. Isn't block import always going to fail, because blob sidecars are not imported.

Reply:
I changed the way I push data in the tracker. Now i put them separated and the import should wait until the tracker completes.
This is what I have in mind, need to test if it actually behaves this way:)

Reply:
this is actually working.